spec/git_helper_spec.rb

```diff
@@ -90,4 +90,61 @@
       Fastlane::Helper::GitHelper.commit(message: @message, push: true)
     end
   end
+
+  describe '#is_ignored?' do
+    before do
+      allow(FastlaneCore::Helper).to receive(:sh_enabled?).and_return(true)
+    end
+
+    let(:path) { 'dummy.txt' }
+
+    it 'returns false when the path is not ignored' do
+      setup_git_repo(
+        dummy_file_path: path,
+        add_file_to_gitignore: false
+      )
+      expect(Fastlane::Helper::GitHelper.is_ignored?(path: path)).to be false
+    end
+
+    context 'when the path is in the .gitignore' do
+      # rubocop:disable RSpec/ExampleLength
+      it 'returns true when the .gitignore has uncommitted changes' do
+        # For some reason, I was expecting the underlying `git check-ignore`
+        # command to fail in this case, but I'm clearly wrong.
+        #
+        # I think there's value in keeping this behavior explicity documented
+        # and verified here. – Gio
+        setup_git_repo(
+          dummy_file_path: path,
+          add_file_to_gitignore: true,
+          commit_gitignore: false
+        )
+        expect(Fastlane::Helper::GitHelper.is_ignored?(path: path)).to be true
+      end
+
+      it 'returns true when the .gitignore has no uncommitted changes' do
+        setup_git_repo(
+          dummy_file_path: path,
+          add_file_to_gitignore: true,
+          commit_gitignore: true
+        )
+        expect(Fastlane::Helper::GitHelper.is_ignored?(path: path)).to be true
+      end
+      # rubocop:enable RSpec/ExampleLength
+    end
+  end
+end
+
+def setup_git_repo(dummy_file_path:, add_file_to_gitignore:, commit_gitignore: false)
+  `git init`
+  `touch .gitignore`
+  `git add .gitignore && git commit -m 'Add .gitignore'`
+
+  `echo abc > #{dummy_file_path}`
+
+  # no point in commiting the gitignore if the file shouldn't be in it
+  return unless add_file_to_gitignore
+
+  `echo #{dummy_file_path} > .gitignore`
+  `git add .gitignore && git commit -m 'Update .gitignore'` if commit_gitignore
```

Comment:
First time I read this I thought "it would be nice and safer if all this was done in a temporary directory". Then I went back to read the whole spec file and saw that the tmpdir switch is done via `before` and `after` on the whole spec, so should be ok after all, just thought I'd leave that comment around so that other reviewers aren't tricked by same train of thought 😛

Reply:
When I wrote the first version of this test, it took me at least 5 minutes or beard scratching to figure out why it wasn't working, which was because of the temp dir being set in the `before`.

There's some lessons here for me, nothing new, but stuff that bears repeating:

- Always read (or at least skim) a test file before starting to work on it
- Beware of global setup code, it's sneaky and easy to overlook

Reply:
> There's some lessons here for me, nothing new, but stuff that bears repeating:

you forgot:

3. Keep your beard, it's always handy to have it around to be able to scratch it.